# Exercise 4.6.a Combining and Exporting IC Data

This script contains the following:

1. Importing libraries
2. Checking dimensions
3. Merging df_ords & df_ords_prior
4. Export to pkl file

# 01. Importing Libraries

In [1]:
# Importing libraries

import pandas as pd
import numpy as np
import os

In [2]:
# Path to IC folders
path = r"C:\Users\alabaro\OneDrive - DAI\Documents\Aug_2023 Instacart Basket Analysis"

In [3]:
# importing orders_products_prior
df_ords_prior = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders_products_prior.csv'), index_col = False)

In [4]:
# importing orders cvs
df_ords = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders.csv'), index_col = False)

# 02. Checking Dimensions

In [6]:
# checking output

df_ords_prior.head(5)

order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

In [7]:
# checking output

df_ords.head(5)

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [8]:
df_ords_prior.shape

(32434489, 4)

In [9]:
df_ords.shape

(3421083, 7)

# 03. Merge df_ords & df_ords_prior

In [10]:
# Merge df_ords & df_ords_prior

df_merged_large = df_ords.merge(df_ords_prior, on = 'order_id', indicator = True)

In [12]:
# check output

df_merged_large.head()

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2539329        1    prior             1          2                  8   
2   2539329        1    prior             1          2                  8   
3   2539329        1    prior             1          2                  8   
4   2539329        1    prior             1          2                  8   

   days_since_prior_order  product_id  add_to_cart_order  reordered _merge  
0                     NaN         196                  1          0   both  
1                     NaN       14084                  2          0   both  
2                     NaN       12427                  3          0   both  
3                     NaN       26088                  4          0   both  
4                     NaN       26405                  5          0   both

In [15]:
df_merged_large['_merge'].value_counts()

_merge
both          32434489
left_only            0
right_only           0
Name: count, dtype: int64

The new merged file has 32,434,489 rows

# 04. Export new file to pkl

In [16]:
df_merged_large.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders_products_combined.pkl'))